modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/message/body/MultipartBodyMessageBodyReader.java

```diff
@@ -99,35 +127,31 @@ public MultipartBody readFrom(
 
 		if (binaryFiles.isEmpty() && values.isEmpty()) {
 			try {
-				ServletFileUpload servletFileUpload = new ServletFileUpload(
-					new DiskFileItemFactory());
+				Map<String, FileItem[]> multipartParameterMap =
+					uploadServletRequest.getMultipartParameterMap();
 
-				List<FileItem> fileItems = servletFileUpload.parseRequest(
-					_httpServletRequest);
+				for (Map.Entry<String, FileItem[]> entry :
+						multipartParameterMap.entrySet()) {
 
-				for (FileItem fileItem : fileItems) {
-					String name = fileItem.getFieldName();
+					FileItem fileItem = entry.getValue()[0];
 
-					if (fileItem.isFormField()) {
-						values.put(
-							name, Streams.asString(fileItem.getInputStream()));
-					}
-					else {
-						binaryFiles.put(
-							name,
-							new BinaryFile(
-								fileItem.getContentType(), fileItem.getName(),
-								fileItem.getInputStream(), fileItem.getSize()));
-					}
+					binaryFiles.put(
+						entry.getKey(),
+						new BinaryFile(
+							fileItem.getContentType(), fileItem.getFileName(),
+							fileItem.getInputStream(), fileItem.getSize()));
 				}
-			}
-			catch (FileUploadBase.SizeLimitExceededException
-						sizeLimitExceededException) {
 
-				throw new BadRequestException(
-					"Please enter a file with a valid file size no larger " +
-						"than " + DLValidatorUtil.getMaxAllowableSize(0, null),
-					sizeLimitExceededException);
+				Map<String, List<String>> regularParameterMap =
```

Comment:
What is the purpose of this loop? I don't see it in the original code. Why are the regular parameters needed?

Reply:
This is [the code](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/message/body/MultipartBodyMessageBodyReader.java#L112) from the original class. With the new class, the parameters that are not a file are added to the `regularParameterMap`

Reply:
oh I missed that.